docs/proposals/network-policy.md

```diff
@@ -0,0 +1,174 @@
+the namespace is required.  That is accomplished through ingress rules on `NetworkPolicy`
+objects (of which there can be multiple in a single namespace).  Pods selected by
+a NetworkPolicy should allow any incoming connections that match any
+ingress rule, per the network plugin’s capabilities.
+
+```yaml
+kind: NetworkPolicy
+apiVersion: v1beta 
+metadata:
+  name:
+  namespace:
+spec:
+  podSelector:            // Standard label selector - selects pods.  
+  ingress:                // List of ingress rules (optional).             
+    - ports:              // List of allowed ports / protocols (optional).          
+        - port:           // Port on the specified protocol (optional). 
+          protocol:       // Protocol (TCP, UDP) 
+      from:               // List of allowed sources (optional).    
+        - pods:           // Label selector - selects Pods (optional). 
+          namespaces:     // Label selector - selects Namespaces (optional).
+```
+
+Each NetworkPolicy object supports the following fields:
+- `podSelector`: A label selector which selects which pods this `NetworkPolicy` applies to.
+- `ingress`: A list of ingress rules.  If not present, this `NetworkPolicy` has no effect on ingress to the selected pods.
+
+Each ingress rule supports the following fields:
+- `ports`: A list of protocol / ports which should be accessible on the selected pods. If not defined, then access to the selected pods will not be restricted by port/protocol.
+- `from`: A list of source criteria which dictate which sources should be able to access the pods selected. If no `from` key is provided, then access will not be restricted based on source.
+
+Each item in the `ports` list supports the following fields:
+- `protocol`: The protocol for the given port - TCP or UDP.
+- `port`: The port on the specified protocol.
+
+Each item in the `from` list may declare at most one of the following:
+- `pods`: A label selector which selects pods (in the same namespace as the NetworkPolicy).
+- `namespaces`: A label selector which selects namespaces.
+
+Note:
+- Any traffic which does not match one of the NetworkPolicy ingress rules will be dropped via a default
+drop behavior (this occurs whether or not network-isolation has been enabled on the Namespace).
+- All ingress rules are whitelist rules, meaning that it should be easy to resolve the case where multiple
+NetworkPolicy objects select the same set of pods, as there can be no conflicting rules.
+- All pods will always be accessible from the host that they are running on.  This is required to allow for kubelet health checks.
+
+### Go Representation
+
+```go
+type NetworkPolicy struct {
+  TypeMeta
+  ObjectMeta
+
+  Spec NetworkPolicySpec 
+}
+
+type NetworkPolicySpec struct {
+  // Selects the pods which this NetworkPolicy object
+  // applies to.
+  PodSelector map[string]string `json:"podSelector,omitempty"`
```

Comment:
Must be new-style label selector. See Job, ReplicaSet, Deployment, or DaemonSet.